consensus/dummy/dynamic_fees.go

```diff
@@ -37,22 +37,37 @@ var (
 
 // CalcExtraPrefix takes the previous header and the timestamp of its child
 // block and calculates the expected extra prefix for the child block.
-//
-// CalcExtraPrefix should only be called if [timestamp] >= [config.ApricotPhase3Timestamp]
 func CalcExtraPrefix(
 	config *params.ChainConfig,
 	parent *types.Header,
 	timestamp uint64,
 ) ([]byte, error) {
-	window, err := calcFeeWindow(config, parent, timestamp)
-	return window.Bytes(), err
+	switch {
+	case config.IsApricotPhase3(timestamp):
+		window, err := calcFeeWindow(config, parent, timestamp)
+		return window.Bytes(), err
```

Comment:
nit just to be safe, in case i.e. window is nil now or in the future (+wrap error perhaps - up to you)

```suggestion
		if err != nil {
			return nil, fmt.Errorf("failed to calculate fee window: %w", err)
		}
		return window.Bytes(), nil
```